### PR TITLE
Fully editable agnostic

### PR DESCRIPTION
**Pull request by DavidGeismarLtd**
Now we can edit any model with the same set of methods

### CODE REVIEW

File: app/models/editable_fields_collection.rb [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+  end
   private
 
-  def to_hash(editable_fields)
+  def to_hash(editables_config)
     hash = {}
-    editable_fields.each do |field|
-      hash[field.label] = ::DataEditable.new(field)
+    editables_config.each do |param, editables|
+      hash[editables.first.class.to_s.underscore] ||= {}
```
Comment (ManyBaby) [inline app/models/editable_fields_collection.rb:19]: This line is really fragile, Using the safe navigation operator would be a small improvement but this is not really robust anyway
```suggestion
      hash[editables&.first&.class&.to_s&.underscore] ||= {}
```

File: app/views/contact_form_submissions/new.html.erb [hunk 2]
```diff
@@ -1,18 +1,18 @@
 <% title 'Contact' %>
 <div class="boxed">
-  <h1 class="heading1 mb-5 <%= editable_text_class %>" <%= data_editables @editable_fields['page_title']  %>><%= raw @editable_fields['page_title'].value %></h1>
+  <h1 class="heading1 mb-5 <%= editable_text_class %>" <%= data_editables @editables['page_title']  %>><%= raw @editables['page_title'].value %></h1>
```
Comment (ManyBaby) [inline app/views/contact_form_submissions/new.html.erb:3]: I still don't get the importance of data_editables.
My understanding is that they will never change. so why do we need to store them in db anyway.
I'm totally missing the point here.

Comment (DavidGeismarLtd) [reply app/views/contact_form_submissions/new.html.erb:3]: I dont store them in the db, I just use this method to generate various data-attributes : `data-editablemodel, data-editableparam, data-editableroute`, I then use those attributes on the JS side to construct the related update query

File: app/models/editable_fields_collection.rb (hunk 1 above)
Comment (DavidGeismarLtd) [reply app/models/editable_fields_collection.rb:19]: you only need & before first cause everything in ruby is a class !

**Pull request merged**